### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by tnmyshukla**

### CODE REVIEW

File: src/main/java/com/Assignment/AssignmentFour/entity/Item.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  }
+  /**
+   *Method to get final price.
+   * @return final price
+   */
+  public double getFinalPrice() {
+    return finalPrice;
+  }
+
+  public void setFinalPrice() {
```
Comment (varunvora) [inline src/main/java/com/Assignment/AssignmentFour/entity/Item.java]: abstract this logic out of the entity. Can use builder pattern.

File: derby.log [hunk 2]
```diff
@@ -0,0 +1,13 @@
+----------------------------------------------------------------
```
Comment (varunvora) [inline derby.log]: is this needed?

File: src/main/java/com/Assignment/AssignmentFour/service/ItemService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+@Service
+public class ItemService {
+  @Autowired
+  ItemRepository itemRepository;
+  Logger logger= LoggerFactory.getLogger(ItemService.class);
+  @Async("taskExecutor")
+  public CompletableFuture<List<Item>> getItems() {
+    logger.info("get list of user by "+Thread.currentThread().getName());
+    List<Item>allItems=new ArrayList<>();
+    itemRepository.findAll().forEach(allItems::add);
```
Comment (varunvora) [inline src/main/java/com/Assignment/AssignmentFour/service/ItemService.java]: redundant code

File: src/main/java/com/Assignment/AssignmentFour/service/ItemService.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+//    return Items.stream().filter(t->t.getId().equals(id)).findFirst().get();
+  }
+  @Async("taskExecutor")
+  public void addItem(Item item){
+    logger.info("add new user by "+Thread.currentThread().getName());
+    itemRepository.save(new Item(item.getName(), item.getType(), item.getPrice(), item.getQuantity()));
+  }
+  @Async("taskExecutor")
+  public void updateItem(Item item, String id) {
```
Comment (varunvora) [inline src/main/java/com/Assignment/AssignmentFour/service/ItemService.java]: id not used correctly.

File: src/main/java/com/Assignment/AssignmentFour/service/ItemService.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    itemRepository.save(new Item(item.getName(), item.getType(), item.getPrice(), item.getQuantity()));
```
Comment (varunvora) [inline src/main/java/com/Assignment/AssignmentFour/service/ItemService.java]: best practice: updated item should be returned

File: src/main/java/com/Assignment/AssignmentFour/controller/ItemController.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  /**
+   * Instance of service.
+   */
+  private ItemService itemService;
+
+  @RequestMapping("/items")
+  /**
+   * Method to get all items.
+   */
+  public CompletableFuture<ResponseEntity<List<Item>>> getAllItems() {
```
Comment (varunvora) [inline src/main/java/com/Assignment/AssignmentFour/controller/ItemController.java]: not sure if completablefuture makes any difference here

**Review by varunvora — CHANGES_REQUESTED** [review]
Need to test the multithreading logic

File: src/main/java/com/Assignment/AssignmentFour/entity/Item.java (hunk 1 above)
Comment (tnmyshukla) [reply src/main/java/com/Assignment/AssignmentFour/entity/Item.java]: Created itembuilder class.

**Pull request merged**